tests/testcore/onebox.go

```diff
@@ -656,6 +659,20 @@ func (c *TemporalImpl) frontendConfigProvider() *config.Config {
 	}
 }
 
+func (c *TemporalImpl) configProvider(serviceName primitives.ServiceName) func() *config.Config {
+	return func() *config.Config {
+		return &config.Config{
```

Comment:
I'd write this like:
```suggestion
func (c *TemporalImpl) configProvider(serviceName primitives.ServiceName) *config.Config {
	return &config.Config{
```
and then just `fx.Provide(c.configProvider)` above. The fx graph already has a `primitives.ServiceName` corresponding to the service.